primitives/src/valid_proof_of_investor.rs

```diff
@@ -17,8 +15,16 @@ impl ValidProofOfInvestor {
     pub fn evaluate_claim(claim: &Claim, id: &IdentityId, proof: &InvestorZKProofData) -> bool {
         match claim {
             Claim::InvestorUniqueness(Scope::Ticker(ticker), scope_id, cdd_id) => {
-                let message = InvestorZKProofData::make_message(id, ticker);
-                Self::verify_proof(cdd_id, id, scope_id, ticker, proof, &message)
+                let message = InvestorZKProofData::make_message(id, ticker.as_slice());
+                Self::verify_proof(cdd_id, id, scope_id, ticker.as_slice(), proof, &message)
+            }
+            Claim::InvestorUniqueness(Scope::Identity(identity), scope_id, cdd_id) => {
+                let message = InvestorZKProofData::make_message(id, &identity.to_bytes());
+                Self::verify_proof(cdd_id, id, scope_id, &identity.to_bytes(), proof, &message)
+            }
+            Claim::InvestorUniqueness(Scope::Custom(scope), scope_id, cdd_id) => {
```

Comment:
We don't need `Identity` & `Custom` as scope as we are only supporting `Ticker` as scope for the `InvestorUniqueness` claim.

Reply:
True, although ideally we'd generalise this on the compliance rule side as well (allow asset issuers to define the scope / granularity they want to enforce uniqueness across). This would be useful if they wanted unique entity counts across multiple assets etc. - there didn't seem to be any reason to not support this on the InvestorUniqueness side.

Reply:
True there is no reason to argue about this but the point here is that this claim is an investor and asset-specific so this claim will never be used across multiple assets so it is more of an extra blob of code in the codebase.